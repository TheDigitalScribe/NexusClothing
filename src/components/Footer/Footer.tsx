import React from "react"
import Link from "next/link";

export const Footer: React.FC = () => {
  const currentYear = new Date().getFullYear();

  return (
    <div className="py-2 px-4 sm:py-4 sm:px-8 flex">
      <div className="container mx-auto items-center space-y-4 text-center">
        <Link href="/about-us">
          <h3 className="text-base sm:text-md mb-4 transition-colors duration-200 hover:text-blue-700">About Us</h3>
        </Link>
        <Link href="/contact=us">
          <h3 className="text-base sm:text-md transition-colors duration-200 hover:text-blue-700">Contact Us</h3>
        </Link>
        <p className="text-sm text-center">Â© {currentYear} Nexus Clothing. All rights reserved.</p>
      </div>
    </div>
  )
}
