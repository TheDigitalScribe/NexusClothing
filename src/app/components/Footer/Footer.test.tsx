import { render, screen } from '@testing-library/react';
import { Footer } from './Footer';

// Mock next/link
jest.mock('next/link', () => ({
  __esModule: true,
  default: ({ children, href }: { children: React.ReactNode; href: string }) => (
    <a href={href}>{children}</a>
  ),
}));

describe('Footer', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      render(<Footer />);
    });

    it('matches snapshot', () => {
      const { container } = render(<Footer />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Content', () => {
    beforeEach(() => {
      render(<Footer />);
    });

    it('renders "About Us" link', () => {
      const aboutLink = screen.getByText('About Us');
      expect(aboutLink).toBeInTheDocument();
      expect(aboutLink.closest('a')).toHaveAttribute('href', '/about-us');
    });

    it('renders "Contact Us" link', () => {
      const contactLink = screen.getByText('Contact Us');
      expect(contactLink).toBeInTheDocument();
      expect(contactLink.closest('a')).toHaveAttribute('href', '/contact=us');
    });

    it('renders copyright text with current year', () => {
      const currentYear = new Date().getFullYear();
      const copyright = screen.getByText(`© ${currentYear} Nexus Clothing. All rights reserved.`);
      expect(copyright).toBeInTheDocument();
    });
  });

  describe('Styling', () => {
    it('applies correct classes to container', () => {
      const { container } = render(<Footer />);
      const outerDiv = container.firstChild as HTMLElement;
      expect(outerDiv).toHaveClass('py-2', 'px-4', 'sm:py-4', 'sm:px-8', 'flex');
    });

    it('applies correct classes to inner container', () => {
      const { container } = render(<Footer />);
      const innerDiv = container.querySelector('.container');
      expect(innerDiv).toHaveClass('container', 'mx-auto', 'items-center', 'space-y-4', 'text-center');
    });

    it('applies correct classes to headings', () => {
      const { container } = render(<Footer />);
      const headings = container.getElementsByTagName('h3');
      Array.from(headings).forEach(heading => {
        expect(heading).toHaveClass('text-base', 'sm:text-md', 'font-bold');
      });
    });

    it('applies correct classes to copyright text', () => {
      render(<Footer />);
      const copyright = screen.getByText(/© .* All rights reserved\./);
      expect(copyright).toHaveClass('text-sm', 'text-center');
    });
  });

  describe('Accessibility', () => {
    it('has correct heading elements', () => {
      const { container } = render(<Footer />);
      const headings = container.getElementsByTagName('h3');
      expect(headings.length).toBe(2);
    });

    it('has accessible links', () => {
      const { container } = render(<Footer />);
      const links = container.getElementsByTagName('a');
      expect(links.length).toBe(2);
      Array.from(links).forEach(link => {
        expect(link).toHaveAttribute('href');
      });
    });
  });

  describe('Dynamic Content', () => {
    it('updates copyright year automatically', () => {
      const mockDate = new Date('2025-01-01');
      const spy = jest
        .spyOn(global, 'Date')
        .mockImplementation(() => mockDate as unknown as Date);

      render(<Footer />);
      expect(screen.getByText(/© 2025/)).toBeInTheDocument();

      spy.mockRestore();
    });
  });

  describe('Layout', () => {
    it('maintains correct spacing between elements', () => {
      const { container } = render(<Footer />);
      const aboutHeading = screen.getByText('About Us');
      expect(aboutHeading).toHaveClass('mb-4');
    });

    it('centers content horizontally', () => {
      const { container } = render(<Footer />);
      const innerContainer = container.querySelector('.container');
      expect(innerContainer).toHaveClass('mx-auto', 'text-center');
    });
  });

  describe('Edge Cases', () => {
    it('handles long text without breaking layout', () => {
      const mockDate = new Date(2024, 0, 1);
      jest.spyOn(global, 'Date')
        .mockImplementation(() => mockDate);

      const { container } = render(<Footer />);

      const copyright = screen.getByText(/© 2024/);
      expect(copyright).toBeVisible();
      expect(container.firstChild).toBeVisible();

      jest.restoreAllMocks();
    });

    it('handles very long company name', () => {
      const { container } = render(<Footer />);
      const copyright = container.querySelector('p');
      expect(copyright).toHaveClass('text-sm');
      expect(copyright).toBeVisible();
    });

    it('maintains layout with minimum content', () => {
      const { container } = render(<Footer />);
      expect(container.firstChild).toHaveClass('py-2', 'px-4');
    });
  });

});
