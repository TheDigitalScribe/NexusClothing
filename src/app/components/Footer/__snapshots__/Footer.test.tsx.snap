// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Rendering matches snapshot 1`] = `
<div>
  <div
    class="py-2 px-4 sm:py-4 sm:px-8 flex"
  >
    <div
      class="container mx-auto items-center space-y-4 text-center"
    >
      <a
        href="/about-us"
      >
        <h3
          class="text-base sm:text-md font-bold mb-4"
        >
          About Us
        </h3>
      </a>
      <a
        href="/contact=us"
      >
        <h3
          class="text-base sm:text-md font-bold"
        >
          Contact Us
        </h3>
      </a>
      <p
        class="text-sm text-center"
      >
        Â© 
        2024
         Nexus Clothing. All rights reserved.
      </p>
    </div>
  </div>
</div>
`;
